app/assets/javascripts/admin/multiple_input_field_handler.js

```diff
@@ -0,0 +1,43 @@
+function MultipleInputFieldHandler(options) {
+  this.button = options.button;
+  this.container = options.container;
+  this.hiddenInputDiv = options.hiddenInputDiv;
+}
+
+MultipleInputFieldHandler.prototype.addInputFieldToContainer = function() {
+  var inputDiv = this.hiddenInputDiv.clone().removeClass("hidden-input-div"),
+      _this = this;
+  inputDiv.find("[data-type='add-more-field']").each(function(index, input) {
+    $input = $(input);
+    $input.attr('name', _this.returnNameFieldValue($input));
+  });
+  this.container.append(inputDiv);
+}
+
+MultipleInputFieldHandler.prototype.returnNameFieldValue = function(input) {
+  return input.data("model") + '[' + input.data("association") + '][' + $.now() + ']';
```

Comment:
store $.now in a variable outside loop, as there may be a case when we are saving multiple fields here.